### PR TITLE
CI setup that runs tests e2e testing using GPU machines

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
CI setup that runs tests using GPU machines:
- Builds and installs habitat-sim with ` python setup.py install` and `./build.sh --headless`
- Runs `pytest` tests on habitat-sim with testing data
- Added speed regression test with FPS threshold (900 FPS) , thanks @erikwijmans for advice how to do it in elegant way.
- Added run of habitat-api tests on master branch. That will guarantee that HSIM changes won't break HAPI
- Added pytorch installation and running habitat baseline test in prediction mode. That maybe too much of testing, but decided to keep it for now.

Have ideas how to speed  up and parallelize jobs. Will follow up in next PRs.

## How Has This Been Tested
Tests time estimations:
```
Install conda and dependencies 02:56
Install cuda 04:32
Build, install habitat-sim and run benchmark 03:13
Run sim tests 03:08
Run api tests with pytorch 08:14
```

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          command: |
+              curl -o ~/miniconda.sh -O  https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+              chmod +x ~/miniconda.sh
+              ~/miniconda.sh -b -p $HOME/miniconda
+              rm ~/miniconda.sh
+              $HOME/miniconda/bin/conda install -y numpy pyyaml scipy ipython mkl mkl-include
+              export PATH=$HOME/miniconda/bin:$PATH
+              conda create -y -n habitat python=3.6
+              . activate habitat
+              conda install -y numpy pyyaml scipy ipython mkl mkl-include pytest
```
Comment (erikwijmans) [inline .circleci/config.yml]: Let's add a `conda install -y -c conda-forge ninja` -- This should help with build speed

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      - run:
+          name: Run sim tests
+          command: |
+              cd habitat-sim
+              export PATH=$HOME/miniconda/bin:$PATH
+              . activate habitat
+              python examples/example.py --scene data/scene_datasets/habitat-test-scenes/van-gogh-room.glb
+              export PYTHONPATH=$PYTHONPATH:$(pwd)
+              ./build.sh --headless
+              pip install -e .
```
Comment (erikwijmans) [inline .circleci/config.yml]: Possibly bad things will happen as the pip install isn't headless.

```suggestion
              HEADLESS=True pip install -e .
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          name: Install cuda
+          no_output_timeout: 60m
+          command: |
+              wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/cuda-repo-ubuntu1604_8.0.44-1_amd64.deb
+              sudo dpkg -i cuda-repo-ubuntu1604_8.0.44-1_amd64.deb
+              sudo apt-get update || true
+              sudo apt-get --yes --force-yes install cuda
+              nvidia-smi
+      - run:
+          name: Install habtiat-sim
```
Comment (msavva) [inline .circleci/config.yml]: typo: `habtiat` -> `habitat`

**Comment by mathfac** [comment]
- Added speed regression test, thanks @erikwijmans for advice how to do it in elegant way
- Added run of habitat-api tests on master branch. That will guarantee that HSIM changes won't break HAPI
- Added pytorch installation and running habitat baseline test in prediction mode. That maybe too much of testing, but decided to keep it for now.

**Comment by mathfac** [comment]
CI tests passed, merging.

**Pull request merged**